cli/k8s/kubectl.go

```diff
@@ -16,6 +18,7 @@ type Kubectl interface {
 	StartProxy(potentialErrorWhenStartingProxy chan error, port int) error
 	UrlFor(namespace string, extraPathStartingWithSlash string) (*url.URL, error)
 	ProxyPort() int
+	SelfCheck() ([]healthcheck.CheckResult, error)
```

Comment:
what do you think about replacing this line with `healthcheck.StatusChecker` ? to be more explicit that we're implementing this interface?

Reply:
Done. I've been known for getting too carried away wit structural typing with reasonable performance 🙄